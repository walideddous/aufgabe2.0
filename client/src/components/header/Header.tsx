import React, { Fragment } from 'react';
import { Col, PageHeader, Popconfirm, Button, Space } from 'antd';
import { UnorderedListOutlined } from '@ant-design/icons';

//Import types
import { TManagedRoute } from '../../types/types';

interface Tprops {
  clickedHeaderButton: string;
  isHeaderSaveButtonDisabled: boolean;
  currentManagedRoute: TManagedRoute | undefined;
  onClearAll: () => void;
  onCLickOnHeaderSaveButton: () => void;
  onClickOnHeaderLoadButton: () => void;
  onClickOnHeaderNewButton: () => void;
  onClickOnHeaderCancelButton: () => void;
  onLoadManagedRouteQuery: (modes: string[], key: string) => void;
  onDeleteManagedRouteMutation: (key: string) => void;
}

const Header = ({
  clickedHeaderButton,
  isHeaderSaveButtonDisabled,
  currentManagedRoute,
  onClearAll,
  onCLickOnHeaderSaveButton,
  onClickOnHeaderLoadButton,
  onClickOnHeaderNewButton,
  onClickOnHeaderCancelButton,
  onLoadManagedRouteQuery,
  onDeleteManagedRouteMutation,
}: Tprops) => {
  return (
    <Fragment>
      {clickedHeaderButton === 'laden' ||
      clickedHeaderButton === 'erstellen' ? (
        <Col xs={24}>
          <PageHeader
            title={
              <div
                style={{
                  color: '#4383B4',
                  fontSize: '17px',
                  display: 'flex',
                  justifyContent: 'space-between',
                }}
              >
                <UnorderedListOutlined
                  style={{ paddingTop: '9px', paddingRight: '10px' }}
                />
                <p>
                  {clickedHeaderButton
                    ? `Linie ${clickedHeaderButton}`
                    : `Linie manager`}
                </p>
              </div>
            }
            style={{ padding: '16px' }}
            extra={[
              //@ts-ignore
              <Space key='1' wrap>
                {clickedHeaderButton === 'laden' && currentManagedRoute && (
                  <>
                    <Button
                      type='primary'
                      htmlType='submit'
                      form='formWrapper'
                      disabled={isHeaderSaveButtonDisabled}
                      onClick={() => {
                        onCLickOnHeaderSaveButton();
                      }}
                    >
                      Speichern
                    </Button>
                    <Popconfirm
                      title={`Wollen Sie wirklich die Linie löschen ?`}
                      placement='bottomRight'
                      okText='Ja'
                      cancelText='Nein'
                      onConfirm={() => {
                        if (currentManagedRoute) {
                          const { key } = currentManagedRoute;
                          onDeleteManagedRouteMutation(key);
                        }
                      }}
                    >
                      <Button danger id='delete_stopSequence'>
                        löschen
                      </Button>
                    </Popconfirm>
                    <Popconfirm
                      title={`Wollen Sie wirklich die Linie resetten ?`}
                      placement='bottomRight'
                      okText='Ja'
                      cancelText='Nein'
                      onConfirm={() => {
                        if (currentManagedRoute) {
                          const { key, modes } = currentManagedRoute;
                          onClearAll();
                          onLoadManagedRouteQuery(modes, key);
                        }
                      }}
                    >
                      <Button danger id='delete_stopSequence'>
                        Reset
                      </Button>
                    </Popconfirm>
                  </>
                )}
                {clickedHeaderButton === 'erstellen' && (
                  <>
                    <Button
                      type='primary'
                      htmlType='submit'
                      form='formWrapper'
                      disabled={isHeaderSaveButtonDisabled}
                      onClick={() => {
                        onCLickOnHeaderSaveButton();
                      }}
                    >
                      Speichern
                    </Button>
                  </>
                )}
                <Popconfirm
                  title={`Wollen Sie wirklich nicht mehr die Linie ${clickedHeaderButton} ?`}
                  placement='bottomRight'
                  okText='Ja'
                  cancelText='Nein'
                  onConfirm={() => {
                    onClickOnHeaderCancelButton();
                  }}
                >
                  <Button type='dashed'>Abbrechen</Button>
                </Popconfirm>
              </Space>,
            ]}
          />
        </Col>
      ) : (
        <Col xs={24}>
          <PageHeader
            title={
              <div
                style={{
                  color: '#4383B4',
                  fontSize: '17px',
                  display: 'flex',
                  justifyContent: 'space-between',
                }}
              >
                <UnorderedListOutlined
                  style={{ paddingTop: '9px', paddingRight: '10px' }}
                />
                <p>
                  {clickedHeaderButton
                    ? `Linie ${clickedHeaderButton}`
                    : `Linie manager`}
                </p>
              </div>
            }
            style={{ padding: '16px' }}
            extra={[
              <Space key='1'>
                <Button
                  id='load_button'
                  value='laden'
                  onClick={() => {
                    onClickOnHeaderLoadButton();
                  }}
                >
                  Laden
                </Button>
                <Button
                  id='new_button'
                  value='erstellen'
                  onClick={() => {
                    onClickOnHeaderNewButton();
                  }}
                >
                  Neu
                </Button>
              </Space>,
            ]}
          />
        </Col>
      )}
    </Fragment>
  );
};

export default Header;
