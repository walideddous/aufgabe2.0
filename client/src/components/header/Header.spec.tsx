import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Header from './Header';

const makeProps = (props: any) => ({
  radioButton: '',
  saveButtonDisabled: false,
  currentManagedRoute: {},
  onClearAll() {},
  onSaveButton() {},
  onLadenButton() {},
  onErstellenButton() {},
  onAbbrechenButton() {},
  onDisplayStopSequenceQuery() {},
  onDeleteStopSequenceMutation() {},
  ...props,
});

const setUp = (props: any) => {
  const shallowWrapper = shallow(<Header {...props} />);
  return shallowWrapper;
};

describe('Test the Header Component', () => {
  let shallowWrapper: any;

  const clickedHeaderButton = '';
  const isHeaderSaveButtonDisabled = true;
  const currentManagedRoute = {};
  const onClearAll = jest.fn();
  const onCLickOnHeaderSaveButton = jest.fn();
  const onClickOnHeaderLoadButton = jest.fn();
  const onClickOnHeaderNewButton = jest.fn();
  const onClickOnHeaderCancelButton = jest.fn();
  const onDisplayStopSequenceQuery = jest.fn();
  const onDeleteStopSequenceMutation = jest.fn();

  beforeEach(() => {
    shallowWrapper = setUp(
      makeProps({
        clickedHeaderButton,
        isHeaderSaveButtonDisabled,
        currentManagedRoute,
        onClearAll,
        onCLickOnHeaderSaveButton,
        onClickOnHeaderLoadButton,
        onClickOnHeaderNewButton,
        onClickOnHeaderCancelButton,
        onDisplayStopSequenceQuery,
        onDeleteStopSequenceMutation,
      })
    );
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('Should test the jest framework', () => {
    expect(true).toBe(true);
  });
  it('Should match the Header component snapshot', () => {
    expect(toJSON(shallowWrapper)).toMatchSnapshot();
  });
  it('Should test  ');
});
