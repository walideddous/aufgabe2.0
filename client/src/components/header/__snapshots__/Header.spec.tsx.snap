// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Header Component Should match the Header component snapshot 1`] = `
<Fragment>
  <Col
    xs={24}
  >
    <PageHeader
      extra={
        Array [
          <Space>
            <ForwardRef(InternalButton)
              id="load_button"
              onClick={[Function]}
              value="laden"
            >
              Laden
            </ForwardRef(InternalButton)>
            <ForwardRef(InternalButton)
              id="new_button"
              onClick={[Function]}
              value="erstellen"
            >
              Neu
            </ForwardRef(InternalButton)>
          </Space>,
        ]
      }
      style={
        Object {
          "padding": "16px",
        }
      }
      title={
        <div
          style={
            Object {
              "color": "#4383B4",
              "display": "flex",
              "fontSize": "17px",
              "justifyContent": "space-between",
            }
          }
        >
          <ForwardRef(UnorderedListOutlined)
            style={
              Object {
                "paddingRight": "10px",
                "paddingTop": "9px",
              }
            }
          />
          <p>
            Linie manager
          </p>
        </div>
      }
    />
  </Col>
</Fragment>
`;
